src/multivariate.jl

```diff
@@ -159,57 +178,80 @@ function _logpdf(d::TuringMvLogNormal, x::AbstractMatrix{T}) where {T<:Real}
+MvLogNormal(m::TrackedVector{<:Real}, A::Matrix{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, A))
+MvLogNormal(m::AbstractVector{<:Real}, A::TrackedMatrix{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, A))
 
 # dense mean, diagonal covariance
 function MvLogNormal(
     m::TrackedVector{<:Real},
     D::Diagonal{T, <:TrackedVector{T}} where {T<:Real},
 )
-    return MvLogNormal(m, sqrt.(D.diag))
+    return TuringMvLogNormal(TuringMvNormal(m, D))
 end
 function MvLogNormal(
     m::AbstractVector{<:Real},
     D::Diagonal{T, <:TrackedVector{T}} where {T<:Real},
 )
-    return MvLogNormal(m, sqrt.(D.diag))
+    return TuringMvLogNormal(TuringMvNormal(m, D))
 end
 
 # dense mean, diagonal covariance
-MvLogNormal(m::TrackedVector{<:Real}, σ::TrackedVector{<:Real}) = TuringMvLogNormal(TuringDiagMvNormal(m, σ))
-MvLogNormal(m::TrackedVector{<:Real}, σ::AbstractVector{<:Real}) = TuringMvLogNormal(TuringDiagMvNormal(m, σ))
-MvLogNormal(m::TrackedVector{<:Real}, σ::Vector{<:Real}) = TuringMvLogNormal(TuringDiagMvNormal(m, σ))
-MvLogNormal(m::AbstractVector{<:Real}, σ::TrackedVector{<:Real}) = TuringMvLogNormal(TuringDiagMvNormal(m, σ))
+MvLogNormal(m::TrackedVector{<:Real}, σ::TrackedVector{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, σ))
+MvLogNormal(m::TrackedVector{<:Real}, σ::AbstractVector{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, σ))
+MvLogNormal(m::TrackedVector{<:Real}, σ::Vector{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, σ))
+MvLogNormal(m::AbstractVector{<:Real}, σ::TrackedVector{<:Real}) = TuringMvLogNormal(TuringMvNormal(m, σ))
 
 # dense mean, constant variance
 function MvLogNormal(m::TrackedVector{<:Real}, σ::TrackedReal)
-    return TuringMvLogNormal(MvNormal(m, σ))
+    return TuringMvLogNormal(TuringMvNormal(m, σ))
 end
 function MvLogNormal(m::TrackedVector{<:Real}, σ::Real)
-    return TuringMvLogNormal(MvNormal(m, σ))
+    return TuringMvLogNormal(TuringMvNormal(m, σ))
 end
 function MvLogNormal(m::AbstractVector{<:Real}, σ::TrackedReal)
-    return TuringMvLogNormal(MvNormal(m, σ))
+    return TuringMvLogNormal(TuringMvNormal(m, σ))
 end
 
 # dense mean, constant variance
 function MvLogNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:TrackedReal})
-    return TuringMvLogNormal(MvNormal(m, A))
+    return TuringMvLogNormal(TuringMvNormal(m, A))
 end
 function MvLogNormal(m::AbstractVector{<:Real}, A::UniformScaling{<:TrackedReal})
-    return TuringMvLogNormal(MvNormal(m, A))
+    return TuringMvLogNormal(TuringMvNormal(m, A))
 end
 function MvLogNormal(m::TrackedVector{<:Real}, A::UniformScaling{<:Real})
-    return TuringMvLogNormal(MvNormal(m, A))
+    return TuringMvLogNormal(TuringMvNormal(m, A))
 end
 
 # zero mean,, constant variance
-MvLogNormal(d::Int, σ::TrackedReal{<:Real}) = MvLogNormal(zeros(d), σ)
+MvLogNormal(d::Int, σ::TrackedReal{<:Real}) = TuringMvLogNormal(TuringMvNormal(d, σ))
+
+## Zygote adjoint
+
+Zygote.@adjoint function Distributions.MvNormal(
```

Comment:
Using `@adjoint` does not require Zygote but only ZygoteRules in principle, but I'm not sure how much we rely on Zygote-only functions in other places.

Reply:
I think we can depend on ZygoteRules only in the src, but still load Zygote anyways to support Tracker + MvNormal automatically without the users having to load Zygote themselves which would be weird.